### DISCUSSION TITLE
How to send notif via Ntfy but not using httpS, only http (internal docker network)

### DISCUSSION THREAD

**Discussion by kuolemaaa** [containrrr/shoutrrr#452] [Q&A]
I container that I use uses this shoutrrr library but it contact my ntfy instance by https.

The problem is that I want this container to contact my ntfy instance using the docker network without TLS

**Comment by 855579762735** [comment]
Took me a bit to solve this one, but you can accomplish this with a custom script.

I am currently using an app called [Scrutiny](https://github.com/AnalogJ/scrutiny) that utilizes these [shoutrr urls](https://containrrr.dev/shoutrrr/v0.8/services/overview/), so in my `scrutiny.yaml` file I applied the [notify url](https://github.com/AnalogJ/scrutiny/blob/master/example.scrutiny.yaml#L61-L85) like so;
```yaml
version: 1
notify:
  urls:
    - "script:///opt/scrutiny/config/ntfy.sh"
```
and then create a script called `ntfy.sh` under my `config` directory that looks like this;
```sh
#! /bin/bash
curl 'http://ntfy/topic/publish?message=Hard+drive+errors+detected!&priority=high'
```
notice that my curl address is not a domain nor an IP address, it's simply `http://ntfy` .. assuming scrutiny and ntfy share the same docker **network** you can just leave this `ntfy` part as the declared [container_name](https://docs.docker.com/reference/compose-file/services/#container_name), docker's built in DNS will resolve it to the correct IP address in the docker network :)

This worked for me with a docker network that had no internet access to allow notifications to get through. The script I showed here is very basic and gets the job done in notifying you that something is wrong, but is not very descriptive of course .. I plan on making this `ntfy.sh` pass the proper information to ntfy eventually to get a proper notification.

If you create a better script at any point feel free to post it ! hope this helps.

**Comment by piksel** [comment]
You can just append `&scheme=http` to the URL:
https://containrrr.dev/shoutrrr/v0.8/services/ntfy/#:~:text=Scheme%20%2D%20Server%20protocol%2C%20http%20or%20https)
